src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1673,6 +1678,7 @@ def __init__(
         *,
         query_id_place_holder: Optional[str] = None,
         is_ddl_on_temp_object: bool = False,
+        temp_name_place_holder: Optional[Tuple[str, TempObjectType]] = None,
```

Comment:
is it possible that you can have two temp_name_place_holders, like joining two dataframes with temp tables?

Reply:
a join statement when resolve can contain multiple temp_name_place_holders, but currently, when we generate temp object like temp tables or temp file format, we create them one at a time.